--- modules/menu/menu.module	2007-01-05 20:05:54.000000000 +0100
+++ sites/all/modules/localizer/menu/menu.module	2007-04-16 14:52:37.000000000 +0200
@@ -409,8 +409,8 @@ function menu_edit_item_form($type, $mid
 /**
  * Process menu and menu item add/edit form submissions.
  */
-function menu_edit_item_form_submit($form_id, $form_values) {
-  menu_edit_item_save($form_values);
+function menu_edit_item_form_submit($form_id, &$form_values) {
+  $form_values['mid']=menu_edit_item_save($form_values);
   return 'admin/build/menu';
 }
 
