--- block.module.orig	2007-09-15 16:12:02.000000000 +0200
+++ block.module	2007-09-15 16:11:14.000000000 +0200
@@ -1,5 +1,5 @@
 <?php
-// $Id: block.module.patch,v 1.1.2.1 2007/09/15 14:29:13 robertogerola Exp $
+// $Id: block.module.patch,v 1.1.2.1 2007/09/15 14:29:13 robertogerola Exp $
 
 /**
  * @file
@@ -641,7 +641,7 @@ function block_list($region) {
   static $blocks = array();
 
   if (!count($blocks)) {
-    $result = db_query("SELECT DISTINCT b.* FROM {blocks} b LEFT JOIN {blocks_roles} r ON b.module = r.module AND b.delta = r.delta WHERE b.theme = '%s' AND b.status = 1 AND (r.rid IN (%s) OR r.rid IS NULL) ORDER BY b.region, b.weight, b.module", $theme_key, implode(',', array_keys($user->roles)));
+    $result = db_query(db_rewrite_sql("SELECT DISTINCT b.* FROM {blocks} b LEFT JOIN {blocks_roles} r ON b.module = r.module AND b.delta = r.delta WHERE b.theme = '%s' AND b.status = 1 AND (r.rid IN (%s) OR r.rid IS NULL) ORDER BY b.region, b.weight, b.module", 'blocks', 'module'), $theme_key, implode(',', array_keys($user->roles)));
     while ($block = db_fetch_object($result)) {
       if (!isset($blocks[$block->region])) {
         $blocks[$block->region] = array();
